planner/cascades/transformation_rules.go

```diff
@@ -32,12 +34,131 @@ func GetTransformationRules(node plannercore.LogicalPlan) []Transformation {
 }
 
 var transformationMap = map[memo.Operand][]Transformation{
-	/**
-	operandSelect: []Transformation{
-		nil,
+	memo.OperandSelection: {
+		&PushSelDownTableScan{},
+		&PushSelDownTableGather{},
 	},
-	operandProject: []Transformation{
-		nil,
+	memo.OperandDataSource: {
+		&EnumeratePaths{},
 	},
-	*/
+}
+
+// PushSelDownTableScan pushes the selection down to TableScan.
+type PushSelDownTableScan struct {
+}
+
+// GetPattern implements Transformation interface. The pattern of this rule is `Selection -> TableScan`.
+func (r *PushSelDownTableScan) GetPattern() *memo.Pattern {
+	ts := memo.NewPattern(memo.OperandTableScan)
+	return memo.BuildPattern(memo.OperandSelection, ts)
+}
+
+// Match implements Transformation interface.
+func (r *PushSelDownTableScan) Match(expr *memo.ExprIter) bool {
+	return true
+}
+
+// OnTransform implements Transformation interface.
+func (r *PushSelDownTableScan) OnTransform(old *memo.ExprIter) (newExprs []*memo.GroupExpr, eraseOld bool, eraseAll bool, err error) {
+	sel := old.GetExpr().ExprNode.(*plannercore.LogicalSelection)
+	ts := old.Children[0].GetExpr().ExprNode.(*plannercore.TableScan)
```

Comment:
Can we make sure its `Children[0]` always `TableScan`. I found that the pattern-check in `findMoreEquiv` only check the first operand and the `Match` function of this rule always returns true.